src/extensions/default/HealthData/HealthDataManager.js

```diff
@@ -0,0 +1,170 @@
+     */
+    function getHealthData() {
+        var result = new $.Deferred(),
+            oneTimeHealthData = {};
+
+        var userUuid = PreferencesManager.getViewState("UUID");
+
+        if (!userUuid) {
+            userUuid = uuid.v4();
+            PreferencesManager.setViewState("UUID", userUuid);
+        }
+
+        oneTimeHealthData.guid = userUuid;
+        oneTimeHealthData.snapshotTime = (new Date()).getTime();
+        oneTimeHealthData.os = brackets.platform;
+        oneTimeHealthData.userAgent = navigator.userAgent;
+        oneTimeHealthData.osLanguage = brackets.app.language;
+        oneTimeHealthData.bracketsLanguage = brackets.getLocale();
+        oneTimeHealthData.bracketsVersion = brackets.metadata.version;
+
+        HealthDataUtils.getInstalledExtensions()
+            .done(function (userInstalledExtensions) {
+                oneTimeHealthData.installedExtensions = userInstalledExtensions;
+            })
+            .always(function () {
+                return result.resolve(oneTimeHealthData);
+            });
+
+        return result.promise();
+    }
+
+    /**
+     * Send data to the server
+     */
+    function sendHealthDataToServer() {
+        var result = new $.Deferred();
+        
+        getHealthData().done(function (healthData) {
+            var url = brackets.config.healthDataServerURL,
+                data = JSON.stringify(healthData);
+
+            $.ajax({
+                url: url,
+                type: "POST",
+                data: data,
+                dataType: "text",
+                contentType: "text/plain"
+            })
+                .done(function () {
+                    result.resolve();
+                })
+                .fail(function (jqXHR, status, errorThrown) {
+                    console.error("Error in sending health data. Response : " + jqXHR.responseText + ". Status : " + status + ". Error : " + errorThrown);
+                    result.reject();
+                });
+        })
+            .fail(function () {
+                result.reject();
+            });
+
+        return result.promise();
+    }
+
+    /*
+     * Check if health data is to be send to the server. If user has enable the tracking, health data will be send for every 24 hours.
+     * If 24 hours or more than that has been passed, then send health data to the server.
+     * We are sending the data as soon as user has launched the brackets. It will only send when the notification dialog for opt-out/in is closed.
+     */
+    function checkHealthDataSend() {
+        var result = new $.Deferred(),
+            isHDTracking = prefs.get("healthDataTracking"),
+            notificationDialogShown = PreferencesManager.getViewState("healthDataNotificationShown");
+        
+        window.clearTimeout(timeoutVar);
+
+        if (isHDTracking && notificationDialogShown) {
+
+            var lastTimeSent = PreferencesManager.getViewState("lastTimeSentData"),
```

Comment:
View states are usually not being looked at by the user, but `lastTimeSentData` isn't specific to Health Data.
I'd go with either `lastTimeSentHealthData` or `healthData.lastTimeSent` (the dot notation is probably not a bad idea for the other prefs, too. You could also rewrite it to use a `getExtensionPrefs()` context).